documentation/debugging.rst

```diff
@@ -0,0 +1,106 @@
+    more information on the binded classes.
+
+2.  Since the includes from the ``bits`` directory should not appear in the generated code,  one can grep for ``bits`` in the generated codes, 
+   i.e. ``grep -r "bits" cmake_bindings/*`` could yield:
+
+.. code-block:: console
+
+    cmake_bindings/std/complex.cpp:#include <bits/stl_construct.h> // std::_Construct
+    cmake_bindings/std/complex.cpp:#include <bits/stl_construct.h> // std::_Destroy
+    cmake_bindings/std/complex.cpp:#include <bits/stl_construct.h> // std::_Destroy_aux
+    cmake_bindings/std/complex.cpp:#include <bits/stl_construct.h> // std::_Destroy_aux<true>::__destroy
+    cmake_bindings/std/complex.cpp:#include <bits/stl_construct.h> // std::_Destroy_n_aux
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_copy
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_copy<false>::__uninit_copy
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_copy<true>::__uninit_copy
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_copy_a
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_1
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n_1
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n_1<false>::__uninit_default_n
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n_1<true>::__uninit_default_n
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_n_a
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_novalue_1
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_default_novalue_n_1
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_fill
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_fill_n
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::__uninitialized_move_if_noexcept_a
+    cmake_bindings/std/complex.cpp:#include <bits/stl_uninitialized.h> // std::uninitialized_copy
+  
+The important information in the output is the ``std::`` types/functions without the leading underscores.
+ Those are STL-implementation independent types/functions that should be defined elsewhere, not in the headers from the ``bits`` directory.
+In this particular example, the function of interest is ``std::uninitialized_copy``.  
+
+A quick search in the C++ documentation or in the WWW,  tells that this function is defined in the <memory> header.
```

Comment:
let's make "C++ documentation" a link to https://en.cppreference.com which is de facto main C++ documentation right now

Reply:
OK. Done.

Reply:
👍